src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -241,6 +241,75 @@ export default class Fms {
         return routeSegments.join(DIRECT_ROUTE_SEGMENT_SEPARATOR);
     }
 
+    /**
+     * Return the next waypoint which has an altitude restriction
+     *
+     * @for Fms
+     * @property nextAltitudeRestrictedWaypoint
+     * @type {WaypointModel}
+     */
+    get nextAltitudeRestrictedWaypoint() {
+        const waypoints = this.getAltitudeRestrictedWaypoints();
+
+        return waypoints[0];
+    }
+
+    /**
+     * Return the next waypoint which has an "AT" altitude restriction
+     *
+     * @for Fms
+     * @property nextHardAltitudeRestrictedWaypoint
+     * @type {WaypointModel}
+     */
+    get nextHardAltitudeRestrictedWaypoint() {
```

Comment:
Are you using the native `filter` here for performance?

the `filter` should be on a new line, as is the convention with chained methods, and can also take advantage of the `=>` direct return
```javascript
const waypoints = this.getAltitudeRestrictedWaypoints()
    .filter((waypoint) => waypoint.altitudeMaximum === waypoint.altitudeMinimum);
```